### ISSUE TITLE
Regression: MSTest 3.2.0 breaks previously working code with 3.1.1 using UITestMethodAttribute

### ISSUE THREAD

**Issue by abdes** [microsoft/testfx#2188]
## Describe the bug

MSTest v3.2.0 breaks the following code, which now unable to locate the `UITestMethodAttribute` class.

```csharp
using System.Diagnostics.CodeAnalysis;
using Microsoft.UI.Xaml;
using Microsoft.VisualStudio.TestTools.UnitTesting.AppContainer;

public partial class App
{
    /// <summary>
    /// Initializes a new instance of the <see cref="App" /> class.
    /// </summary>
    public App() => this.InitializeComponent();

    protected override void OnLaunched(LaunchActivatedEventArgs args)
    {
        // Ensure the current window is active
        var window = new MainWindow();
        window.Activate();
        UITestMethodAttribute.DispatcherQueue = window.DispatcherQueue;

        // Replace back with e.Arguments when https://github.com/microsoft/microsoft-ui-xaml/issues/3368 is fixed
        Microsoft.VisualStudio.TestPlatform.TestExecutor.UnitTestClient.Run(Environment.CommandLine);
    }
}
```
This code works fine with 3.1.1.

The issue is most likely due to dependencies that you upgraded in MSTest that no longer offer the `UITestMethodAttribute`.

Such **breaking change** is not documented in the release notes.

As this attribute was very important in testing WinUI code, please explain what is the alternative now?

## Steps To Reproduce

1. Create a WinUI simple App
2. Replace App.xaml.cs with teh code above.
3. Add MSTest references
3.1. Use v3.1.1, code compiles 
3.2. Use v3.2.0, code does not compile

## Expected behavior

No breaking changes without explanation and a migration path.

## Actual behavior

Breaking changes that leave us with no other way to simply test UI code.

## Additional context

Visual Studio Community Version 17.8.5

```xml
  <ItemGroup>
    <PackageVersion Include="MSTest.TestAdapter" Version="3.1.1" />
    <PackageVersion Include="MSTest.TestFramework" Version="3.1.1" />
    <PackageVersion Include="FluentAssertions" Version="7.0.0-alpha.3" />
    <PackageVersion Include="coverlet.collector" Version="6.0.0" />
    <PackageVersion Include="coverlet.msbuild" Version="6.0.0" />
    <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.9.0-release-23619-01" />
    <PackageVersion Include="Moq" Version="4.20.70" />
    <PackageVersion Include="Microsoft.TestPlatform.TestHost" Version="17.9.0-release-23619-01" />
    <PackageVersion Include="Verify.DiffPlex" Version="2.3.0" />
    <PackageVersion Include="Verify.MSTest" Version="23.0.1" />
    <PackageVersion Include="Verify.SourceGenerators" Version="2.2.0" />
  </ItemGroup>
```

[AB#1948343](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1948343)

**Comment by Evangelink** [comment]
Hi @abdes,

Thanks and sorry we introduce this regression. We will investigate and get back to you asap.

**Comment by Evangelink** [comment]
@abdes would you mind sharing some minimal repro? When creating a project following https://devblogs.microsoft.com/ifdef-windows/winui-desktop-unit-tests/ I am not able to run the tests at all even with older versions of MSTest.

**Comment by abdes** [comment]
> @abdes would you mind sharing some minimal repro? When creating a project following https://devblogs.microsoft.com/ifdef-windows/winui-desktop-unit-tests/ I am not able to run the tests at all even with older versions of MSTest.
> 
> Here is my project sample (in case you find some obvious wrong parts): [App2.zip](https://github.com/microsoft/testfx/files/14109604/App2.zip)

Thanks @Evangelink . Actually, **your App2.zip runs perfectly fine, both, with 3.1.1 and 3.2.0**.

The issue starts to happen when you move net8.0.
```diff
diff --git a/App2/App2.csproj b/App2/App2.csproj
index 04e029c..d3c1070 100644
--- a/App2/App2.csproj
+++ b/App2/App2.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
        <PropertyGroup>
                <OutputType>WinExe</OutputType>
-    <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
+               <TargetFramework>net8.0-windows10.0.22621.0</TargetFramework>
                <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
                <RootNamespace>App2</RootNamespace>
                <ApplicationManifest>app.manifest</ApplicationManifest>
```

Due to known bug, https://github.com/microsoft/WindowsAppSDK/issues/3842, need to add `<UseRidGraph>true</UseRidGraph>` to get past the errors `error NETSDK1083: The specified RuntimeIdentifier 'win10-x64' is not recognized.`. So that gives us:

```diff
diff --git a/App2/App2.csproj b/App2/App2.csproj
index 04e029c..c0de740 100644
--- a/App2/App2.csproj
+++ b/App2/App2.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
        <PropertyGroup>
                <OutputType>WinExe</OutputType>
-    <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
+               <TargetFramework>net8.0-windows10.0.22621.0</TargetFramework>
                <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
                <RootNamespace>App2</RootNamespace>
                <ApplicationManifest>app.manifest</ApplicationManifest>
@@ -10,6 +10,8 @@
                <PublishProfile>win10-$(Platform).pubxml</PublishProfile>
                <UseWinUI>true</UseWinUI>
                <EnableMsixTooling>true</EnableMsixTooling>
+               <!-- https://github.com/microsoft/WindowsAppSDK/issues/3842 -->
+               <UseRidGraph>true</UseRidGraph>
        </PropertyGroup>
```

With the above changes, 3.1.1, continues to work as previously. 3.2.0 fails to build with:
```
1>F:\projects\mstest-2188\App2\App.xaml.cs(15,52,15,64): error CS0234: The type or namespace name 'AppContainer' does not exist in the namespace 'Microsoft.VisualStudio.TestTools.UnitTesting' (are you missing an assembly reference?)
1>F:\projects\mstest-2188\App2\TestClass.cs(8,52,8,64): error CS0234: The type or namespace name 'AppContainer' does not exist in the namespace 'Microsoft.VisualStudio.TestTools.UnitTesting' (are you missing an assembly reference?)
1>F:\projects\mstest-2188\App2\TestClass.cs(15,10,15,22): error CS0246: The type or namespace name 'UITestMethodAttribute' could not be found (are you missing a using directive or an assembly reference?)
1>F:\projects\mstest-2188\App2\TestClass.cs(15,10,15,22): error CS0246: The type or namespace name 'UITestMethod' could not be found (are you missing a using directive or an assembly reference?)
```

I hope this helps.

**Comment by Evangelink** [comment]
It does! I can reproduce the error! I'll investigate and provide a fix.

**Comment by abdes** [comment]
@Evangelink , A question related to this: is MSTest Runner supposed to work with WinUI test projects? cause I have been trying without success to run tests in projects, similar to the one with used to troubleshoot this PR, but without success. The tests run perfectly fine inside VS 2022.

**Comment by Evangelink** [comment]
Not yet supported! We did some POC internally to confirm what works or not but we didn't get enough time to fully validate that everything is working fine so we decided to not support it. I will update the docs to be clear about that.

For now I don't have any ETA for the support but I do hope we can work on it on the near future.

**Comment by Evangelink** [comment]
@abdes we just released 3.2.1.

**Comment by abdes** [comment]
> @abdes we just released 3.2.1.

and it works perfectly fine.
Thank a lot @Evangelink. I wish all projects have maintainers like you 😄.

**Comment by Evangelink** [comment]
> Thank a lot @Evangelink. I wish all projects have maintainers like you 😄.

Wow that means a lot to me! Much love on you @abdes!
